sql/core/src/main/scala/org/apache/spark/sql/jdbc/MsSqlServerDialect.scala

```diff
@@ -109,7 +109,7 @@ private object MsSqlServerDialect extends JdbcDialect {
         None
       } else {
         sqlType match {
-          case java.sql.Types.SMALLINT => Some(ShortType)
+          case java.sql.Types.SMALLINT | java.sql.Types.TINYINT => Some(ShortType)
```

Comment:
We should map it to `ByteType`

Reply:
Yeah agree. Let me make this change.

Reply:
To make it consistent across all database, can we make it a ShortType instead?

Reply:
> To make it consistent across all database

could you provide more details?

Reply:
For mysql, TinyInt can be a signed value.

Reply:
ByteType is signed too

Reply:
From the doc https://spark.apache.org/docs/latest/sql-ref-datatypes.html, it seems that it cannot represent unsigned value like 255.

Reply:
@zeruibao I am leaving this one to @yaooqinn @MaxGekk @sadikovi

Reply:
Also, in JDBC, is it required to be Symmetrical for `getCatalystType` and `getJDBCType`?

Reply:
I don't think it's Symmetrical.
SMALLINT and TINYTINT -> ShortType
ShortType -> SMALLINT

Reply:
What happens if you use ByteType for TINYINT? Have you tried that? I think it is worth trying it on a few versions of SQL Server with mapping to ByteType or ShortType and see what results you get.

Reply:
I just tried it. Using `ByteType` to read `TinyInt` of 255, I got **-1** instead of **255**. I have to use **ShortType** to get the correct answer.

Reply:
is it true only for TINYINT or all integer types? If INT can be unsigned as well, then shall we map INT to LongType?

Reply:
It's true only for TINYINT.

Reply:
https://learn.microsoft.com/en-us/sql/t-sql/data-types/int-bigint-smallint-and-tinyint-transact-sql?view=sql-server-ver16
You can refer to this doc. Thank you.

Reply:
interesting. Can we add a code comment to mention it and reference to the sqlserver doc?

Reply:
Sure, just add a comment!